### PR TITLE
[help editor] bug fixing for filter and column type

### PR DESCRIPTION
**Pull request by kongtiaowang**
This pull request fixes the column default value and filter.
fixed bugs: can't save help content and can't filter the content.

### CODE REVIEW

File: SQL/0000-00-04-Help.sql [hunk 1]
```diff
@@ -10,64 +10,11 @@ CREATE TABLE `help` (
     `parentID` int(11) NOT NULL DEFAULT '-1',
     `hash` varchar(32) DEFAULT NULL,
     `topic` varchar(100) NOT NULL DEFAULT '',
-    `content` text NOT NULL,
+    `content` text DEFAULT NULL,
```
Comment (driusan) [inline SQL/0000-00-04-Help.sql]: What is this trying to fix? I don't see why we'd want to start allowing null help content. If there's no help, it shouldn't be in the help table.
e

Comment (kongtiaowang) [reply SQL/0000-00-04-Help.sql]: When saving help content for the instrument, it got this error [Insert statement did not execute successfully: Array ( [0] => HY000 [1] => 1364 [2] => Field 'content' doesn't have a default value )].

**Comment by kongtiaowang** [comment]
@driusan 
Could u review again, please?

File: modules/help_editor/test/help_editorTest.php [hunk 2]
```diff
@@ -40,12 +40,12 @@ function setUp()
...
-             'parentID' => '-1',
-             'hash'     => $md5String,
-             'topic'    => 'Test Topic',
-             'content'  => 'This is a test content.',
-             'created'  => '2013-04-05 00:00:00',
+             'helpID'         => '999999',
+             'parentID'       => '-1',
+             'hash'           => $md5String,
+             'topic'          => 'Test Topic',
+             'projectContent' => 'This is a test content.',
```
Comment (driusan) [inline modules/help_editor/test/help_editorTest.php]: I think you removed the wrong column.. it should be `content` that remains, not `projectContent`.

Comment (zaliqarosli) [reply modules/help_editor/test/help_editorTest.php]: ^ i agree with @driusan, every new help content goes into the 'projectContent' column but it should be in 'content'. For every existing entry, it looks like 'projectContent' has value null.

**Comment by zaliqarosli** [comment]
Test comments:
1. projectContent column in help table should be dropped
2. in all calls to update()/insert() the help content in php/edit_help_content.class.inc, all 'projectContent' be changed to 'content' so that the help content is stored in 'content' column of help table
3. this is assuming we are allowing all modules' help content to come from the module_name.md file in each module's directory, and only instruments are to use the help editor module

**Comment by kongtiaowang** [comment]
@driusan @zaliqarosli 
The column name has renamed projectContent as content.

**Comment by driusan** [comment]
I added the caveat tag because content previously updated into the `projectContent` column will be lost, but I'm not sure of any way around that.

File: SQL/0000-00-04-Help.sql [hunk 3]
```diff
@@ -10,64 +10,10 @@ CREATE TABLE `help` (
     `parentID` int(11) NOT NULL DEFAULT '-1',
     `hash` varchar(32) DEFAULT NULL,
     `topic` varchar(100) NOT NULL DEFAULT '',
-    `content` text NOT NULL,
-    `projectContent` text DEFAULT NULL,
+    `content` text DEFAULT NULL,
```
Comment (driusan) [inline SQL/0000-00-04-Help.sql]: I still don't see why this is changed to nullable

**Pull request merged**